src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -263,7 +362,14 @@ static void OnResolve(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t inter
     if (!(flags & kDNSServiceFlagsMoreComing))
     {
         VerifyOrReturn(sdCtx->HasInterface(), sdCtx->Finalize(kDNSServiceErr_BadState));
-        GetAddrInfo(sdCtx);
+
+        // If a resolve was not requested on this context, call GetAddrInfo and set the isResolveRequested flag to true.
+        if (!sdCtx->isResolveRequested)
+        {
+            GetAddrInfo(sdCtx);
+            sdCtx->isResolveRequested        = true;
+            sdCtx->hasOpenThreadTimerStarted = false;
```

Comment:
Why are we setting hasOpenThreadTimerStarted false here?  Shouldn't it be set that way in the constructor?

Reply:
This is incorrect. I removed it before seeing your comment :D. I realized it was wrong